src/btree/bt_ret.c

```diff
@@ -49,6 +41,14 @@ __key_return(WT_CURSOR_BTREE *cbt)
             return (0);
         }
 
+        /* If the cursor references a WT_INSERT item, take its key. */
+        if (cbt->ins != NULL) {
+            WT_ASSERT(CUR2S(cbt), cbt->compare != 0);
```

Comment:
I don't think you need this assert, we can't get here if `cbt.compare == 0` because of the test on line 34?

Reply:
Thanks. Fixed in [fdf5f9b](https://github.com/wiredtiger/wiredtiger/pull/7777/commits/fdf5f9b26904816071e3220be441ad5dc68c41a1)